src/commands/SinceCommand.ts

```diff
@@ -0,0 +1,275 @@
+    const reason: string | undefined = reasonParts?.join(" ");
+
+    for (let targetRoomId of rooms) {
+        let {html, text} = await (async () => {
+            switch (action) {
+                case Action.Show: {
+                    return makeJoinStatus(mjolnir, targetRoomId, maxEntries, minDate, maxAgeMS);
+                }
+                case Action.Kick: {
+                    const joins = mjolnir.roomJoins.getUsersInRoom(targetRoomId, minDate, maxEntries);
+                    let results = { good: 0, bad: 0};
+                    for (let join of joins) {
+                        try {
+                            await mjolnir.client.kickUser(join.userId, targetRoomId, reason);
+                            results.good += 1;
+                        } catch (ex) {
+                            LogService.warn("SinceCommand", "Error while attempting to kick user", ex);
+                            results.bad += 1;
+                        }
+                    }
+                    const text_ = `Attempted to kick ${joins.length} users from room ${targetRoomId}, ${results.good} kicked, ${results.bad} failures`;
+                    return {
+                        html: text_,
+                        text: text_,
+                    }
+                }
+                case Action.Ban: {
+                    const joins = mjolnir.roomJoins.getUsersInRoom(targetRoomId, minDate, maxEntries);
+
+                    let results = { good: 0, bad: 0};
+                    for (let join of joins) {
+                        try {
+                            await mjolnir.client.banUser(join.userId, targetRoomId, reason);
+                            results.good += 1;
+                        } catch (ex) {
+                            LogService.warn("SinceCommand", "Error while attempting to ban user", ex);
+                            results.bad += 1;
+                        }
+                    }
+                    const text_ = `Attempted to ban ${joins.length} users from room ${targetRoomId}, ${results.good} kicked, ${results.bad} failures`;
+                    return {
+                        html: text_,
+                        text: text_
+                    }
+                }
+            }
+        })();
+
+        const reply = RichReply.createFor(destinationRoomId, event, text, html);
+        reply["msgtype"] = "m.notice";
+        /* no need to await */ mjolnir.client.sendMessage(destinationRoomId, reply);
+    }
+
+    await mjolnir.client.redactEvent(destinationRoomId, progressEventId);
+    return {ok: undefined};
+}
+
+function makeJoinStatus(mjolnir: Mjolnir, targetRoomId: string, maxEntries: number, minDate: Date, maxAgeMS: number): {html: string, text: string} {
+    const HUMANIZER_OPTIONS = {
+        // Reduce "1 day" => "1day" to simplify working with CSV.
+        spacer: "",
+        // Reduce "1 day, 2 hours" => "1.XXX day" to simplify working with CSV.
+        largest: 1,
+    };
```

Comment:
Hmm ok, but this does look kind of weird 

![image](https://user-images.githubusercontent.com/50846879/157243765-288d8e40-16a4-4ce6-8684-578b27aeb49e.png)


I guess if we are really providing a CSV option then it is ok.

Reply:
Alternatively, I could probably quote these durations.

Reply:
I think the durations that are listed with the users are ok like this because they are in the same format as when you send the command, and that kind of consistency is good.